lib/Feature/FeatureModelWriter.cpp

```diff
@@ -338,9 +338,16 @@ int FeatureModelXmlWriter::writeFeature(xmlTextWriterPtr Writer,
     }
   }
 
-  // location?
-  if (FeatureSourceRange *Fsr = Feature1.getFeatureSourceRange()) {
-    RC = writeSourceRange(Writer, *Fsr);
+  // locations?
+  if (Feature1.hasLocations()) {
+    RC = xmlTextWriterStartElement(Writer, XmlConstants::LOCATIONS);
+    CHECK_RC
+    auto Locations = Feature1.getLocations();
```

Comment:
Is auto here a ref/ptr/value? Maybe we should put an explicit type here or add add `&` or `*` after the auto.

Reply:
This is a iterator range for the loop below. I removed the local variale and just call Feature.getLocations() in the loop header.